src/services/dex.js

```diff
@@ -1030,37 +1030,72 @@ export default {
     return new Promise((resolve, reject) => {
       try {
         if (assetId === assets.NEO || assetId === assets.GAS) {
+          const systemWithdraw = {
+            step: 0,
+            asset: assetId === assets.NEO ? 'NEO' : 'GAS',
+            amount: quantity.toString(),
+          };
+
+          store.commit('setSystemWithdraw', systemWithdraw);
+
+          store.commit('setWithdrawInProgressModalModel', {
+          });
+
+          const rejectWithError = (errorMsg) => {
+            store.commit('setSystemWithdrawMergeState', { error: errorMsg });
+            reject(errorMsg);
+          };
+          // TODO: should pass this through the whole way instead of getting again in case they switch wallets somehow
+          const currentWallet = wallets.getCurrentWallet();
+
           this.markWithdraw(assetId, quantity)
             .then((res) => {
               if (res.success !== true) {
-                reject('Withdraw Mark Step rejected');
+                rejectWithError('Withdraw Mark Step rejected');
                 return;
               }
+              store.commit('setSystemWithdrawMergeState', { step: 2 });
 
               alerts.success('Withdraw Mark Step Relayed. Waiting for confirmation.');
               neo.monitorTransactionConfirmation(res.tx, true)
                 .then(() => {
+                  store.commit('setSystemWithdrawMergeState', { step: 3 });
+                  const dexAddress = wallet.getAddressFromScriptHash(store.state.currentNetwork.dex_hash);
+                  // Must allow funds to be sent again by moving tx outputs to unspent.
+                  neo.applyTxToAddressSystemAssetBalance(dexAddress, res.tx, true);
+
                   setTimeout(() => {
                     this.withdrawSystemAsset(assetId, quantity, res.tx.hash, res.utxoIndex)
                       .then((res) => {
                         if (res.success) {
+                          store.commit('setSystemWithdrawMergeState', { step: 4 });
+
+                          neo.monitorTransactionConfirmation(res.tx, true)
+                            .then(() => {
```

Comment:
Maybe you can flatten this nested 'then...catch'. Try `return neo.monitorTransactionConfirmation(res.tx, true)` on line 1073. Then you should be able to take away a tab on the 'then' on line 1074. I hope that makes sense...

Reply:
I agree; there is a lot that I had flattened in this method already, but it needs more. Can happen in a subsequent PR.